src/Compilers/Core/CodeAnalysisTest/AnalyzerAssemblyLoaderTests.cs

```diff
@@ -495,6 +498,13 @@ private static void VerifyDependencyAssemblies(AnalyzerAssemblyLoader loader, in
                 .GetAssemblies()
                 .Where(x => isInLoadFromContext(loader, x));
 
+            // When debugging, the debugger will load this DLL and that can throw off the debugging 
+            // session so exclude it here.
+            if (Debugger.IsAttached)
+            {
+                loadedAssemblies = loadedAssemblies.Where(x => x.GetName().Name != "Microsoft.VisualStudio.Debugger.Runtime.Desktop");
+            }
+
```

Comment:
You had a similar assertion elsewhere, but it didn't have the debug check. Should we just have a little "assert nothing new in the ALC" helper? (can defer after the PR merges)

Reply:
This is part of the set of helpers that verify nothing new in the ALC / AppDomain. There is a core helper method and a few others that feed into it. This is one of the core helpers though hence the best place to put this check.